lib/using_restricted_array.rb

```diff
@@ -6,56 +6,115 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+  counter = 0
+  until array[counter] == nil
+    array[counter]
+    counter += 1
+  end
+  return counter
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def print_array(array)
-  raise NotImplementedError
+  counter = 0
+  while counter < length(array)
```

Comment:
Just FYI, this method will call the `length(array)` method over and over again.  So by calling the `length` method every time this loop iterates, you end up with an O(n<sup>2</sup>) algorithm.